### PR TITLE
Visualize probe collections

### PR DESCRIPTION
**Pull request by JoeAtHPI**
Previously, probes were just displayed as pure text. This made it harder to interpret the data fast or find outliners.
image

With this PR, two visualizations were introduced which support number collections and point collections.
image

A visualize button is added to supported collections to toggle a scale and the graph itself. Note that the icon is not final and should be exchanged with bar-chart-2 from https://feathericons.com/?query=chart. The visualizations are aligned over the values themselves to allow a fast backtrack-search in the collection itself to eg inspect the object in question. The user can also hover the visualization points to get the displayed value. 

The visualization encodes information through color.
Red points in the line chart mean that a point is in the upper 10% of the range, blue for the lowest 10%. Same goes for the borders in the rectangle chart. A black border means x/y is low and the other coordinate is high.
Red lines mean we have a decrease between two values, green increase.

### CODE REVIEW

File: packages/Babylonian-UI.package/BPLineChart.class/class/supportedClass.st [hunk 1]
```diff
@@ -0,0 +1,5 @@
+support
+supportedClass
+	
+	
+	^ Number class
```
Comment (codeZeilen) [inline packages/Babylonian-UI.package/BPLineChart.class/class/supportedClass.st:5]: That should probably only be Number.

File: packages/Babylonian-UI.package/BPRectangleChart.class/class/supportedClass.st [hunk 2]
```diff
@@ -0,0 +1,5 @@
+support
+supportedClass
+	
+	
+	^ Point class
```
Comment (codeZeilen) [inline packages/Babylonian-UI.package/BPRectangleChart.class/class/supportedClass.st:5]: Same here should only be point

**Review by codeZeilen — APPROVED** [review]

**Comment by codeZeilen** [comment]
@JoeAtHPI not a request, just a question: is there a reason why the visualization does not use the trace directly?

**Comment by JoeAtHPI** [comment]
@codeZeilen "@JoeAtHPI not a request, just a question: is there a reason why the visualization does not use the trace directly?" do you mean that BPVisualization would take the trace and an id and internally assign `aTrace valuesForProbeId: self id`  as its values? or that one can access the probe's values anytime w/o needing to call `aTrace valuesForProbeId: self id` by the stack? for the second case: i didn't find this more direct way myself. for the first: i thought it'd be nicer for the BPVisualization to decouple itself from needing a trace as it contains far more information than the BPVisualization really needs. i wanted its API to take the mere basics for modularity (while still naming the variables traces related as this is the use case rn)

File: packages/Babylonian-UI.package/BPRectangleChart.class/class/supportedClass.st (hunk 2 above)
Comment (JoeAtHPI) [reply packages/Babylonian-UI.package/BPRectangleChart.class/class/supportedClass.st:5]: If changed, I have to update BPVisualization > visualizationClassesForValues: traceValues from
`containedClasses allSatisfy: [:aTracedValueClass | aTracedValueClass isKindOf: aSubclass supportedClass]` to `containedClasses allSatisfy: [:aTracedValueClass | aTracedValueClass isKindOf: aSubclass supportedClass class]` and `aSubclass supportedClass class` sounded weird for me. No strong feelings here though, what do you prefer?

**Comment by codeZeilen** [comment]
In addition to replacing Point class with Point this should be:
[:aTracedValueClass
| aTracedValueClass withAllSuperclasses includes: supportedClass]

Note that #isKindOf: tests for something different than you intended. It
checks whether the class is of the same type of class, which is probably
not what you meant. It is the result of the Metaclass inheritance hierarchy:

a Point - instanceOf -> Point
Point - subclassOf -> Object
Point - instanceOf -> Point class
Point class - subclassOf -> Object class
!!! Point class - instanceOf -> Metaclass !!!

The last one is used for isKindOf: checks:

```
isKindOf: aClass
  "Answer whether the class, aClass, is a superclass or class of the
receiver."
   ^ self class == aClass or: [ self class inheritsFrom: aClass ]
```


Am Mi., 28. Apr. 2021 um 01:09 Uhr schrieb Joana Be. <
***@***.***>:

**Comment by JoeAtHPI** [comment]
Worked in the feedback. Pipeline seems to be failing for the same reason it's failing on develop

**Comment by codeZeilen** [comment]
Yes, its annoying. One of the dependencies can not be resolved correctly
-.-'

Am Di., 4. Mai 2021 um 17:40 Uhr schrieb Joana Be. ***@***.***

**Comment by JoeAtHPI** [comment]
Green pipeline, nice, so is it good to go? :D

**Pull request merged**